config/cmake/modules/MfemCmakeUtilities.cmake

```diff
@@ -100,6 +100,8 @@ macro(add_mfem_examples EXE_SRCS)
 
     string(REPLACE ".cpp" "" EXE_NAME "${EXE_PREFIX}${SRC_FILENAME}")
     mfem_add_executable(${EXE_NAME} ${SRC_FILE})
+    install(TARGETS ${EXE_NAME}
+            RUNTIME DESTINATION examples)
```

Comment:
Shouldn't this be conditional, depending on `MFEM_ENABLE_EXAMPLES`? Note that this macro is called even if `MFEM_ENABLE_EXAMPLES` is false.

Reply:
Good question @v-dobrev.  The code behaves as expected but there may be an underlying/subtle issue.  My interpretation of what is happening is below.  Barring any further guidance, I'm fine with either adding the conditional around `install(...)` (to be on the safe side), or leaving it as is.

AFAIK, the setting of `EXCLUDE_FROM_ALL` to true here: 

https://github.com/mfem/mfem/pull/2406/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR556

, which was in the base version of this PR, prevents the `install` rule from tracking the example targets, hence the added conditional and call to `add_subdirectory(examples)`.  

Looking at the documentation for `EXCLUDE_FROM_ALL`:

https://cmake.org/cmake/help/v3.20/prop_tgt/EXCLUDE_FROM_ALL.html#prop_tgt:EXCLUDE_FROM_ALL
> If a target has EXCLUDE_FROM_ALL set to true, it may still be listed in an install(TARGETS) command, but the user is responsible for ensuring that the target's build artifacts are not missing or outdated when an install is performed.

and `install`:

https://cmake.org/cmake/help/v3.20/command/install.html
> Installing a target with the EXCLUDE_FROM_ALL target property set to TRUE has undefined behavior.

I suspect that we are avoiding these two potentially problematic cases as we are not setting the individual example target properties as `EXCLUDE_FROM_ALL=true`.

Note, building with `MFEM_ENABLE_EXAMPLES` set to false, I could find no `make` incantation that would force installation of one of the compiled examples (i.e, `make install ex1`, `make examples install`, etc..).

Reply:
> Note, building with `MFEM_ENABLE_EXAMPLES` set to false, I could find no `make` incantation that would force installation of one of the compiled examples (i.e, `make install ex1`, `make examples install`, etc..).


Maybe `make install` from the `examples` directory will unconditionally install the examples?

In any case, if `make install` from the main directory does not install the examples when `MFEM_ENABLE_EXAMPLES` is false, even if the examples were built separately from the library (e.g. with `make examples`) then it should be fine to leave the above `install(...)` as it is.

Reply:
> In any case, if make install from the main directory does not install the examples when MFEM_ENABLE_EXAMPLES is false, even if the examples were built separately from the library (e.g. with make examples) then it should be fine to leave the above install(...) as it is.

Tried `make config prefix=`... then `make examples`. Then separately 'make -j` did not see examples getting compiled again and then `make install` did not install examples. All with `MFEM_ENABLE_EXAMPLES=OFF`.

Reply:
In mfem/build using cmake with `MFEM_ENABLE_EXAMPLES=OFF`, then `make -j`, `make examples`, then `make install`, examples are not installed.

Reply:
| Maybe make install from the examples directory will unconditionally install the examples?

Good call @v-dobrev .  `make install` from the `examples` dir does build and install the examples when `MFEM_ENABLE_EXAMPLES=OFF`.  This seems like acceptable behavior to me.

Reply:
I also tried turning `MFEM_ENABLE_EXAMPLES=ON` then make. Then turn if to `OFF` before `make install`. Then `make install`. Examples are not installed in the target directory.

Reply:
Okay, based on the above observations, I think we can leave the `install(...)` command as it is.